compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs

```diff
@@ -857,12 +857,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 Ok(r)
             })
             .or_else(|error| {
-                let guar = self
-                    .dcx()
-                    .span_delayed_bug(span, "method resolution should've emitted an error");
                 let result = match error {
                     method::MethodError::PrivateMatch(kind, def_id, _) => Ok((kind, def_id)),
-                    _ => Err(guar),
+                    _ => {
```

Comment:
please revert this change. I want to make sure we always fail if this returns `Err` and with your change this code could theoretically end up returning `Ok` for a private method and compilation may still succeed as we mess up reporting an error elsewhere.

Reply:
ok